### DISCUSSION TITLE
Channel mapping

### DISCUSSION THREAD

**Discussion by BillPurvis** [FluidSynth/fluidsynth#767] [Q&A] [answered]
I've recently come across odd behaviour with Fluidsynth. I'm on Ubuntu 20.04 using the versions provided by the Ubuntu repo. I;m running it using it as a Calf Plugin. I have an ancient home-grown sequencer which I wrote many moons ago, updated to use
Jack. Playing a midifile which has 5 music tracks (String quartet, so violin*2,Viola and Cello.

This worked fine, a little shaky at times, but good enough for me. When the sequencer send Program Changes to Fluid
the appear as channels 2-5. What I actually send is 0-3 (as my code was based on an ancient MIDI protocol spec)
which used to map onto Fluid channels 1-4. Is this a new bug? I have monitored the jack output and it confirms the
PC events specify 0-3 so it's not some new bug crept into my sequencer. The notes sound OK, so they must be being
mapped in the same way. Not a great problem but had me confused for a while/

**Comment by derselbst** [comment]
Pls. start fluidsynth directly from console in verbose mode to verify that the prog changes really appear on the "wrong" channels:

```
fluidsynth -j -a jack --verbose
```

Make sure to establish a MIDI connection with, e.g. qjackctl. You should see a bunch of `prog chan bank preset` messages on stdout as they arrive. If they show the wrong  channels, run it in dump mode:

```
fluidsynth -j -a jack --dump
```

to see the midi events before they arrive in the MIDI router. If they still show the wrong channels, you have a mapping problem somewhere else.

**Comment by BillPurvis** [comment, marked as answer]
OK, Looks like the problem is in the Calf GUI.
